### DISCUSSION TITLE
Default values not applying

### DISCUSSION THREAD

**Discussion by larsbaunwall** [eclipse-langium/langium#1333] [Q&A] [answered]
I am trying to add a default value to a property, as implemented in #1165 and tested in https://github.com/eclipse-langium/langium/blob/2ca2b3adb452ad5048135af85e1510f84a71fd7b/packages/langium/test/parser/langium-parser-builder.test.ts#L651

However, trying the same approach on 2.1.0 returns the parsing error `Expecting token of type '}' but found '='.` on an interface declaration in my grammar that looks like this:

```
interface model {
    name: string
    description: string = "hello"
}
```

I am trying to embed documentation into [my grammar](https://github.com/larsbaunwall/DomainLang/blob/main/dsl/dlang/src/language/domain-lang.langium) that I can use in a hoverProvider to explain terms/syntax. 
Maybe that is best kept out of the grammar, or what is your take on that?

**Comment by msujew** [comment, marked as answer]
@larsbaunwall The change is assigned to [milestone v3](https://github.com/eclipse-langium/langium/milestone/11), which will be released some time later this month. The feature isn't available for general use yet.

>I am trying to embed documentation into [my grammar](https://github.com/larsbaunwall/DomainLang/blob/main/dsl/dlang/src/language/domain-lang.langium) that I can use in a hoverProvider to explain terms/syntax.
Maybe that is best kept out of the grammar, or what is your take on that?

Regarding that: I wouldn't recommend using default values for that. Documentation on grammar rules can be extracted from the `CommentProvider` even when the grammar has been serialized. Just use the normal JSDoc syntax (`/** ... */`) to write your documentation.

**Reply by larsbaunwall** [reply]
Ah, I had a hunch, but since it was merged into main I thought it was already in the wild.
Thx!

**Reply by larsbaunwall** [reply]
@msujew thanks for the tip on JSDoc. I also tried that, but the comments don't seem to end up in the `generated/ast` nor the `generated/grammar`?

E.g., using a comment like this in my grammar:

```
/**
 * Partnership relationship
 */
PartnershipRelationship:
    name='<=>'
;
```

Do I have to configure something to make that happen?

**Reply by msujew** [reply]
... I just noticed that we haven't configured our generator to actually generate the comments. You can probably use dummy properties for that for now. In 3.0 we will have that fixed. Thanks for the heads-up.
